### DISCUSSION TITLE
Quantity with decimals

### DISCUSSION THREAD

**Discussion by stefanvanherwijnen** [vendurehq/vendure#681] [General]
Hi,

I am wondering what the best way would be to implement the quantity as a decimal.
For example, this would be useful for selling by weight. The obvious workaround is to enter the quantity in grams, but not everything can be easily divided in smaller fractions (e.g. time,1.5 day is 36 hours but it is more confusing).

Changing the quantity to a floating number might cause calculation issues so I wonder if there is a different solution?

https://barn2.co.uk/sell-by-weight-woocommerce/

**Comment by michaelbromley** [comment]
You could try this:

* add a customField to the OrderLine entity of type 'float'. This will store the decimal quantity.
* In your storefront you provide an input where that decimal quantity can be specified by the customer
* When they add to order, you call `addItemToOrder` and include the custom field containing their decimal quantity
* You define an [OrderItemPriceCalculationStrategy](https://www.vendure.io/docs/typescript-api/orders/order-item-price-calculation-strategy/) which uses the decimal quantity custom field to calculate the price.

Let me know if that would cover your use-case.

**Comment by stefanvanherwijnen** [comment]
Thanks. I was thinking about writing a plugin but didn't notice the OrderItemPriceCalculationStrategy and thought it was hard-coded.
I'll have to figure out the exact implementation but this definitely is the better solution.

Another feature which I think could be helpful is a way to implement small variations in price based on product weight. E.g. I sell a ProductVariant '1kg of apples' but when packaging it turns out to be 1,05 kg. But this is definitely an edge case and is better off in a plugin.

Great work on Vendure so far :+1: .

**Comment by bischofa-code** [comment]
Hi Stefan
as I'm currently struggeling with the same (the apple problem), I wonder how you solved it or whether, in the end, you wrote a (accessible?) plugin. best

**Reply by stefanvanherwijnen** [reply]
I never got around to it unfortunately.

**Reply by michaelbromley** [reply]
There's a related issue here:

- https://github.com/vendure-ecommerce/vendure/issues/2725
